generators/entity-server/index.js

```diff
@@ -101,4 +103,42 @@ module.exports = class extends BaseBlueprintGenerator {
         });
         return { importApiModelProperty, importJsonIgnore, importJsonIgnoreProperties, importSet, uniqueEnums };
     }
+
+    _getJoinColumnName(relationship) {
+        if (relationship.useJPADerivedIdentifier === true) {
+            return 'id';
+        }
+        return `${this.getColumnName(relationship.relationshipName)}_id`;
+    }
+
+    _generateSqlSafeName(name) {
+        if (isReservedTableName(name, 'sql')) {
+            return `e_${name}`;
+        }
+        return name;
+    }
+
+    _getUseMapsIdRelation(relationships) {
+        const jpaDerivedRelation = Object.values(relationships).find(rel => rel.useJPADerivedIdentifier === true);
+        if (jpaDerivedRelation) {
+            return { isUsingMapsId: true, mapsIdAssoc: jpaDerivedRelation };
+        }
+        return { isUsingMapsId: false, mapsIdAssoc: null };
+    }
+
+    _generateEagerRelationsAndEntityTypes(entityClass, relationships) {
+        const eagerRelations = relationships.filter(function (rel) {
+            return rel.relationshipType === 'many-to-one' || (rel.relationshipType === 'one-to-one' && rel.ownerSide === true);
+        });
+        const regularEagerRelations = eagerRelations.filter(function (rel) {
+            return rel.useJPADerivedIdentifier !== true;
+        });
+        const uniqueEntityTypes = new Set(
+            eagerRelations.map(function (rel) {
+                return rel.otherEntityNameCapitalized;
+            })
+        );
+        uniqueEntityTypes.add(entityClass);
+        return { eagerRelations, uniqueEntityTypes, regularEagerRelations };
+    }
```

Comment:
I prefer to do preparation before templates, instead of calling functions there:
```
    _preparing() {
        return {
            setup() {
                this.joinColumName = this. _getJoinColumnName(this.relationships);
                this.derivatedRelationship = this.relationships.find(rel => rel.useJPADerivedIdentifier === true);
                this.isUsingMapsId = !!this.derivatedRelationship;
            },
        };
    }

    get preparing() {
        if (useBlueprints) return;
        return this._preparing();
    }
```

Reply:
Sorry, I was not aware of this complex way of doing template setup. Is the 'preparing()' and setup()' function name is somehow privileged?

Reply:
`preparing` is a custom priority. See https://github.com/jhipster/generator-jhipster/issues/12338.
https://github.com/jhipster/generator-jhipster/blob/9aee4a29381a1ebb52b83f13c8e7083620b5ef9b/generators/entity/index.js#L465-L481

Setup is the task name can be anything.

Reply:
Ok, I see now, thanks!